rust/src/io/object_writer.rs

```diff
@@ -15,29 +15,41 @@
 // specific language governing permissions and limitations
 // under the License.
 
-use std::io::Error;
 use std::pin::Pin;
 use std::task::{Context, Poll};
 
+use object_store::{path::Path, MultipartId};
 use pin_project::pin_project;
 use prost::Message;
 use tokio::io::{AsyncWrite, AsyncWriteExt};
 
 use crate::format::ProtoStruct;
+use crate::io::ObjectStore;
+use crate::Result;
 
 /// AsyncWrite with the capability to tell the position the data is written.
 ///
 #[pin_project]
-pub struct ObjectWriter<'a> {
+pub struct ObjectWriter {
+    store: ObjectStore,
+
     // TODO: wrap writer with a BufWriter.
     #[pin]
-    writer: &'a mut (dyn AsyncWrite + Unpin + Send),
+    writer: Box<dyn AsyncWrite + Unpin + Send>,
+    multipart_id: MultipartId,
     cursor: usize,
 }
 
-impl<'a> ObjectWriter<'a> {
```

Comment:
curious why you got rid of all of the lifetimes

Reply:
Seems easier to "create an object" with the internal reference as copy  and pass around . 

could revisit it if the overhead is too big.